osh/builtin_meta.py

```diff
@@ -328,6 +332,46 @@ def Run(self, cmd_val):
         return 0
 
 
+class Error(vm._Builtin):
+
+    def __init__(self, expr_ev):
+        # type: (expr_eval.OilEvaluator) -> None
+        self.expr_ev = expr_ev
+
+    def Run(self, cmd_val):
+        # type: (cmd_value.Argv) -> int
+        blame = loc.Word(cmd_val.arg_locs[0])
+
+        argc = len(cmd_val.argv)
+        if argc != 1:
+            e_usage('Expected 0 command arguments but received %d' % (argc - 1), blame)
+
+        positional = cmd_val.typed_args.positional
+        if len(positional) != 1:
+            e_usage('Expected 1 positional argument but received %d' % len(positional), blame)
+
+        message_val = self.expr_ev.EvalExpr(positional[0], blame)
+        if message_val.tag() != value_e.Str:
+            raise error.InvalidType2(message_val, 'expected a Str', blame)
+        message = cast(value.Str, message_val).s
+
+        named = cmd_val.typed_args.named
+        status = 1
+        for arg in named:
```

Comment:
This checking looks good for now, I like all the tests, though we also have some code to unify

- I added `frontend/typed_args.py`, which is used in some of the builtins
- you added `library/util.py`
- some functions in `ysh/val_ops.py` too, some of which are unused